src/main/kotlin/org/akuleshov7/utils/Credentials.kt

```diff
@@ -0,0 +1,20 @@
+package org.akuleshov7.utils
+
+@Suppress("USE_DATA_CLASS")
+class Credentials(auth: String) {
+    val gitHubUserName: String
+    val gitHubAuthToken: String
+
+    init {
+        auth.let {
+            val splitAuth = auth.split(":")
+            if (splitAuth.size != 2) {
+                """Incorrect value ($auth) is passed to 'auth' property.
+                    | It should contain user and auth-token separated by a colon (':').
+                    | This option is needed to extend the limit of requests.""".trimMargin() logAndExit 8
```

Comment:
I'm not really familiar with how this is usually done, but shouldn't you describe these exit codes somewhere? There are already quite a few of them.

Reply:
yes, will create an issue https://github.com/akuleshov7/git-stat/issues/21